src/user.rs

```diff
@@ -47,6 +48,13 @@ pub struct LoggedIn;
 #[derive(Clone)]
 pub struct Synching {
     rooms: Arc<Mutex<Vec<Box<RoomId>>>>,
+    available_room_ids: Vec<String>,
+}
+
+impl User<Synching> {
+    pub fn has_friendships(&self) -> bool {
+        !self.state.available_room_ids.is_empty()
+    }
```

Comment:
nit: there is a impl block for this state, maybe we should avoid having repeated impl blocks in same file.

Reply:
Moved it to the expected impl